test_scripts/API/VehicleData/commonVehicleData.lua

```diff
@@ -0,0 +1,301 @@
+
+local function checkIfPTSIsSentAsBinary(bin_data)
+  if not (bin_data ~= nil and string.len(bin_data) > 0) then
+    commonFunctions:userPrint(31, "PTS was not sent to Mobile in payload of OnSystemRequest")
+  end
+end
+
+function commonVehicleData.getGetVehicleDataConfig()
+  return {
+    keep_context = false,
+    steal_focus = false,
+    priority = "NONE",
+    default_hmi = "NONE",
+    groups = { "Base-4", "Emergency-1" }
+  }
+end
+
+local function getPTUFromPTS(tbl)
+  tbl.policy_table.consumer_friendly_messages.messages = nil
+  tbl.policy_table.device_data = nil
+  tbl.policy_table.module_meta = nil
+  tbl.policy_table.usage_and_error_counts = nil
+  tbl.policy_table.functional_groupings["DataConsent-2"].rpcs = json.null
+  tbl.policy_table.module_config.preloaded_pt = nil
+  tbl.policy_table.module_config.preloaded_date = nil
+end
+
+local function jsonFileToTable(file_name)
+  local f = io.open(file_name, "r")
+  local content = f:read("*all")
+  f:close()
+  return json.decode(content)
+end
+
+
+local function addParamToRPC(tbl, functional_grouping, rpc, param)
+  local is_found = false
+  local params = tbl.policy_table.functional_groupings[functional_grouping].rpcs[rpc].parameters
+  for _, value in pairs(params) do
+    if (value == param) then is_found = true end
+  end
+  if not is_found then
+    table.insert(tbl.policy_table.functional_groupings[functional_grouping].rpcs[rpc].parameters, param)
+  end
+end
+
+local function ptu(self, app_id, ptu_update_func)
+  local function getAppsCount()
+    local count = 0
+    for _ in pairs(hmiAppIds) do
+      count = count + 1
+    end
+    return count
+  end
+
+  local policy_file_name = "PolicyTableUpdate"
+  local policy_file_path = commonFunctions:read_parameter_from_smart_device_link_ini("SystemFilesPath")
+  local pts_file_name = commonFunctions:read_parameter_from_smart_device_link_ini("PathToSnapshot")
+  local ptu_file_name = os.tmpname()
+  local requestId = self.hmiConnection:SendRequest("SDL.GetURLS", { service = 7 })
+  EXPECT_HMIRESPONSE(requestId)
+  :Do(function()
+      self.hmiConnection:SendNotification("BasicCommunication.OnSystemRequest",
+        { requestType = "PROPRIETARY", fileName = pts_file_name })
+      getPTUFromPTS(ptu_table)
+       local function updatePTU(tbl)
+        for rpc in pairs(tbl.policy_table.functional_groupings["Emergency-1"].rpcs) do
+          addParamToRPC(tbl, "Emergency-1", rpc, "engineOilLife")
+        end
+        tbl.policy_table.app_policies[commonVehicleData.getMobileAppId(app_id)] = commonVehicleData.getGetVehicleDataConfig()
+      end
+      updatePTU(ptu_table)
+      if ptu_update_func then
+        ptu_update_func(ptu_table)
+      end
+      local function tableToJsonFile(tbl, file_name)
+        local f = io.open(file_name, "w")
+        f:write(json.encode(tbl))
+        f:close()
+      end
+      tableToJsonFile(ptu_table, ptu_file_name)
+
+      local event = events.Event()
```

Comment:
Lua panic here because `events` is never set:
```
PANIC: unprotected error in call to Lua API (./test_scripts/API/VehicleData/commonVehicleData.lua:107: attempt to index global 'events' (a nil value))
```